src/microservices/Microservices.IsIdentifiable/Rules/ConsensusRule.cs

```diff
@@ -0,0 +1,90 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using Dicom.Imaging.Codec;
+using Microservices.IsIdentifiable.Failures;
+
+namespace Microservices.IsIdentifiable.Rules
+{
+    /// <summary>
+    /// Rule in which 2 or more subrules must agree on common sections of problem
+    /// </summary>
+    public class ConsensusRule : ICustomRule
+    {
+        /// <summary>
+        /// Rules which must all reach a consesus on classifications
+        /// </summary>
+        public ICustomRule[] Rules {get;set;}
+
+        /// <summary>
+        /// Applies all <see cref="Rules"/> and returns the consensus <see cref="RuleAction"/>.  All rules must agree on the action and at least one index of one word that is a <see cref="FailurePart"/>
+        /// </summary>
+        /// <param name="fieldName"></param>
+        /// <param name="fieldValue"></param>
+        /// <param name="badParts"></param>
+        /// <returns></returns>
+        public RuleAction Apply(string fieldName, string fieldValue, out IEnumerable<FailurePart> badParts)
+        {
+            RuleAction? firstRuleOperation = null;
+            List<FailurePart> parts = new List<FailurePart>();
+
+            foreach(var rule in Rules)
+            {
+                //first time
+                if(firstRuleOperation == null)
+                {
+                    firstRuleOperation = rule.Apply(fieldName,fieldValue,out IEnumerable<FailurePart> newParts);
+                    parts.AddRange(newParts);
+                }
+                else
+                {
+                    //subsequent times
+                    var newOperation = rule.Apply(fieldName,fieldValue,out IEnumerable<FailurePart> newParts);
+
+                    // There is not consensus on whether the value should be classified as bad
+                    if(newOperation != firstRuleOperation)
```

Comment:
I just wondered if this should also check that they != RuleAction.None ?

Reply:
It hink this would give the maximum efficiency: https://github.com/SMI/SmiServices/pull/438/commits/921bfc0e8d7968c55559d2109aea508e5e3ec329

At end of each loop if the action is None then we stop asking anyone else.